src/payment/strategies/masterpass/masterpass-script-loader.ts

```diff
@@ -1,24 +1,55 @@
 import { ScriptLoader } from '@bigcommerce/script-loader';
 
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { PaymentMethod } from '../../../payment';
 import { PaymentMethodClientUnavailableError } from '../../errors';
 
 import { Masterpass, MasterpassHostWindow } from './masterpass';
 
 export default class MasterpassScriptLoader {
+
     constructor(
         private _scriptLoader: ScriptLoader,
         public _window: MasterpassHostWindow = window
     ) {}
 
-    load(testMode?: boolean, locale?: string, checkoutId?: string): Promise<Masterpass> {
-        return this._scriptLoader
-            .loadScript(`https://${testMode ? 'sandbox.' : ''}src.mastercard.com/srci/integration/merchant.js?locale=${locale}&checkoutid=${checkoutId}`)
-            .then(() => {
-                if (!this._window.masterpass) {
-                    throw new PaymentMethodClientUnavailableError();
-                }
+    async load(paymentMethod?: PaymentMethod, locale?: string ): Promise<Masterpass> {
+
+        if (!paymentMethod) {
```

Comment:
can you make paymentMethod mandatory rather than optional? This way you won't need this check. You are already checking  the strategy that paymentMethod exists.

Also, might be better if you pass just exactly what's required (similar to what this before) rather than an entire object. 
```js
interface MasterpassScriptLoaderParams {
  useMasterpassSrc: boolean;
  language: string;
  testMode: boolean;
}

load({ useMasterpassSrc, language, testMode }: MasterpassScriptLoaderParams): Promise<Masterpass> {